src/bus.js

```diff
@@ -0,0 +1,24 @@
+export default {
+    events: {},
+
+    on(name, callback) {
+        if (! this.events.hasOwnProperty(name)) {
+            this.events[name] = []
+        }
+
+        this.events[name].push(callback)
+    },
+
+    emit(name, data = {}) {
+        if (this.events.hasOwnProperty(name)) {
```

Comment:
Any reason this check isn't just `this.events[name]`?

Reply:
Fair point. I prefer using the explicit methods but to be fair, that would work fine.